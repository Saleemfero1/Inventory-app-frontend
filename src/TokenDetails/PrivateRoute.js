import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./AuthContext";

const PrivateRoute = ({ children }) => {
  const { token } = useContext(AuthContext);
  return token ? <>{children}</> : <Navigate to="/" />;
};

export default PrivateRoute;
