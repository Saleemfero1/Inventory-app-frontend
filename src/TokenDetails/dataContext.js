// import React, { createContext, useState } from "react";

// export const DataContext = createContext();

// export const DataProvider = ({ children }) => {
//   const [dashData, setData] = useState([]);

//   const updateData = (newData) => {
//     setData(newData);
//   };

//   return (
//     <DataContext.Provider value={{ dashData, updateData }}>
//       {children}
//     </DataContext.Provider>
//   );
// };
