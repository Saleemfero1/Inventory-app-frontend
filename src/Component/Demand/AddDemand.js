import React, { useState, useEffect, useContext } from "react";
import { useNavigate } from "react-router-dom";
import LocationServices from "../../Service/LocationServices";
import ItemServices from "../../Service/ItemServices";
import DemandServices from "../../Service/DemandServices";
import { ToastContainer, toast } from "react-toastify";
import { AuthContext } from "../../TokenDetails/AuthContext";
import { Button } from "@mui/material";
import Select from "react-select";

export default function AddDemand() {
  const { token } = useContext(AuthContext);
  const queryParameters = new URLSearchParams(window.location.search);
  const id = queryParameters.get("id");
  const [demandTypes] = useState(["PLANNED", "HARDPROMISED"]);
  const navigate = useNavigate();
  const [items, setItems] = useState([]);
  const [locations, setLocations] = useState([]);

  const [demand, setDemand] = useState({
    itemId: "",
    locationId: "",
    demandType: "",
    quantity: "",
  });

  useEffect(() => {
    if (id == -1) {
      return;
    } else {
      DemandServices.getDemandById(
        sessionStorage.getItem("organizationId"),
        id,
        token
      )
        .then((response) => {
          let existDemand = response.data;
          console.log(existDemand);
          setDemand(existDemand);
        })
        .catch((err) => {
          console.log(err);
        });
    }
  }, []);

  useEffect(() => {
    ItemServices.getActiveItems(sessionStorage.getItem("organizationId"), token)
      .then((response) => {
        setItems(response.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  useEffect(() => {
    LocationServices.getLocations(
      sessionStorage.getItem("organizationId"),
      token
    )
      .then((response) => {
        setLocations(response.data);
        console.log(locations);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  const optionItems = items.map((value) => ({
    label: value[Object.keys(value)[0]],
    value: value[Object.keys(value)[0]],
  }));

  const optionLocations = locations.map((value) => ({
    label: value[Object.keys(value)[0]],
    value: value[Object.keys(value)[0]],
  }));

  const saveDemand = (event) => {
    event.preventDefault();
    let newDemand = {
      itemId: demand.itemId,
      locationId: demand.locationId,
      demandType: demand.demandType,
      quantity: demand.quantity,
    };

    if (id == -1) {
      console.log(newDemand);
      DemandServices.createDemand(
        sessionStorage.getItem("organizationId"),
        newDemand,
        sessionStorage.getItem("token")
      )
        .then((response) => {
          toast.success("demand Created!", {
            onClose: () => navigate("/demand"),
          });
        })
        .catch((err) => {
          toast.error(err.response.data.message);
        });
    } else {
      console.log(newDemand);
      DemandServices.updateDemand(
        sessionStorage.getItem("organizationId"),
        id,
        newDemand,
        token
      )
        .then((response) => {
          toast.success("demand updated!", {
            onClose: () => navigate("/demand"),
          });
        })
        .catch((err) => {
          toast.error(err.response.data.message);
        });
    }
  };

  const onChangeItemId = (itemId) => {
    setDemand({ ...demand, itemId: itemId.value });
  };

  const onChangeLocationId = (locationId) => {
    setDemand({ ...demand, locationId: locationId.value });
  };

  const onChangeDemandType = (event) => {
    setDemand({ ...demand, demandType: event.target.value });
  };

  const onChangeQuantity = (event) => {
    setDemand({ ...demand, quantity: event.target.value });
  };

  return (
    <div className="container mt-5">
      <ToastContainer position="bottom-left" />
      <form
        className="mt-3 p-3 shadow addItems"
        onSubmit={saveDemand}
        data-testid="add-demand-form"
      >
        <div className=" row mb-3 ">
          <div className="form-group col-md-6">
            <label htmlFor="itemId" data-testid="itemId">
              Select Item
            </label>
            <Select
              placeholder="Search item"
              options={optionItems}
              value={demand.itemId["itemId"]}
              onChange={onChangeItemId}
            ></Select>
          </div>
          <div className="form-group col-md-6">
            <label htmlFor="LocationId" data-testid="locationId">
              Location Id
            </label>
            <Select
              placeholder="Serach location"
              options={optionLocations}
              value={demand.locationId["locationId"]}
              onChange={onChangeLocationId}
            ></Select>
          </div>
        </div>

        <div className=" row">
          <div className="form-group col-md-6">
            <label htmlFor="demandType" data-testid="demandType">
              Demand Type
            </label>
            <select
              className="form-select"
              aria-label="Default select example"
              value={demand.demandType}
              onChange={onChangeDemandType}
              required
            >
              <option value="">Choose Demand</option>
              {demandTypes.map((demand, index) => (
                <option key={demand}>{demand}</option>
              ))}
            </select>
          </div>

          <div className="form-group col-md-6">
            <label htmlFor="Quantity" data-testid="Quantity">
              Quantity
            </label>
            <input
              type="text"
              className="form-control"
              id="Quantity"
              placeholder="Enter Quantity"
              value={demand.quantity}
              onChange={onChangeQuantity}
              required
            />
          </div>
        </div>

        <Button
          variant="contained"
          className="bg-info text-black"
          type="submit"
          sx={{ mt: 2 }}
        >
          {id == -1 ? "Add Demand" : "Upadte Demand"}
        </Button>
        <Button
          variant="contained"
          color="error"
          sx={{ ml: 2, mt: 2 }}
          onClick={() => navigate("/demand")}
        >
          Cancel
        </Button>
      </form>
    </div>
  );
}
