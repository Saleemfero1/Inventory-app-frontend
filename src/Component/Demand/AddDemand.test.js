import React from "react";
import renderer from "react-test-renderer";

import { render, screen } from "@testing-library/react";
import AddDemand from "./AddDemand";
import { AuthProvider } from "../../TokenDetails/AuthContext";
import { MemoryRouter } from "react-router-dom";
import Demand from "./Demand";
import { AuthContext } from "../../TokenDetails/AuthContext";
import "@testing-library/jest-dom/extend-expect";
import DemandTable from "./DemandTable";

test("renders AddDemand component correctly", () => {
  render(
    <AuthProvider>
      <MemoryRouter>
        <AddDemand />
      </MemoryRouter>
    </AuthProvider>
  );

  expect(screen.getByTestId("itemId")).toBeInTheDocument();
  expect(screen.getByTestId("locationId")).toBeInTheDocument();
  expect(screen.getByTestId("demandType")).toBeInTheDocument();
  expect(screen.getByTestId("Quantity")).toBeInTheDocument();
  expect(screen.getByTestId("add-demand-form")).toMatchSnapshot();
});

test("renders Demand component correctly", () => {
  render(
    <AuthProvider>
      <MemoryRouter>
        <Demand />
      </MemoryRouter>
    </AuthProvider>
  );

  const addDemandButton = screen.getByTestId("addDemand");
  expect(addDemandButton).toBeInTheDocument();
  expect(addDemandButton).toHaveTextContent("Add Demand");
  expect(screen.getByText("Demand Details")).toBeInTheDocument();
  expect(screen.getByTestId("search")).toBeInTheDocument();
  expect(screen.getByTestId("demand")).toMatchSnapshot();
});

test("DemandTable renders correctly", () => {
  const headings = [
    "Demand Id",
    "Item Id",
    "Location Id",
    "Demand Type",
    "Quantity",
  ];
  const demandData = [
    {
      id: 1,
      itemId: "123",
      locationId: "456",
      demandType: "Type A",
      quantity: 10,
    },
    {
      id: 2,
      itemId: "789",
      locationId: "012",
      demandType: "Type B",
      quantity: 5,
    },
  ];
  const deleteFun = jest.fn();
  const updateFun = jest.fn();
  const search = "";

  const component = renderer.create(
    <DemandTable
      headings={headings}
      demandData={demandData}
      deleteFun={deleteFun}
      updateFun={updateFun}
      search={search}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
