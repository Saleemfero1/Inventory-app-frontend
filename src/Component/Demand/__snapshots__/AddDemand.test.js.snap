// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemandTable renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
>
  <table
    aria-label="customized table"
    className="MuiTable-root css-1pclhgl-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Demand Id
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Item Id
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Location Id
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Demand Type
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Quantity
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
          colSpan={2}
          scope="col"
          style={
            Object {
              "fontSize": "17px",
            }
          }
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    />
    <tfoot
      className="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeMedium MuiTablePagination-root css-fikjyc-MuiTableCell-root-MuiTablePagination-root"
          colSpan={12}
        >
          <div
            className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id=":r1:"
            >
              Rows per page:
            </p>
            <div
              className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
              onClick={[Function]}
            >
              <select
                aria-label="rows per page"
                className="MuiNativeSelect-select MuiTablePagination-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                id=":r0:"
                multiple={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={5}
              >
                <option
                  className="MuiTablePagination-menuItem"
                  value={5}
                >
                  5
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={10}
                >
                  10
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={25}
                >
                  25
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={-1}
                >
                  All
                </option>
              </select>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiTablePagination-selectIcon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              0â€“0 of 0
            </p>
            <div
              className="MuiBox-root css-1zye22"
            >
              <button
                aria-label="first page"
                className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FirstPageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                </svg>
              </button>
              <button
                aria-label="previous page"
                className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                  />
                </svg>
              </button>
              <button
                aria-label="next page"
                className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                </svg>
              </button>
              <button
                aria-label="last page"
                className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LastPageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`renders AddDemand component correctly 1`] = `
<form
  class="mt-3 p-3 shadow addItems"
  data-testid="add-demand-form"
>
  <div
    class=" row mb-3 "
  >
    <div
      class="form-group col-md-6"
    >
      <label
        data-testid="itemId"
        for="itemId"
      >
        Select Item
      </label>
      <div
        class=" css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-13cymwt-control"
        >
          <div
            class=" css-16674sh-Component"
          >
            <div
              class=" css-1jqq78o-placeholder"
              id="react-select-2-placeholder"
            >
              Search item
            </div>
            <div
              class=" css-d9r47t-Component"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-ny0e4k-Component"
          >
            <span
              class=" css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="form-group col-md-6"
    >
      <label
        data-testid="locationId"
        for="LocationId"
      >
        Location Id
      </label>
      <div
        class=" css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-13cymwt-control"
        >
          <div
            class=" css-16674sh-Component"
          >
            <div
              class=" css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            >
              Serach location
            </div>
            <div
              class=" css-d9r47t-Component"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-ny0e4k-Component"
          >
            <span
              class=" css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class=" row"
  >
    <div
      class="form-group col-md-6"
    >
      <label
        data-testid="demandType"
        for="demandType"
      >
        Demand Type
      </label>
      <select
        aria-label="Default select example"
        class="form-select"
        required=""
      >
        <option
          value=""
        >
          Choose Demand
        </option>
        <option>
          PLANNED
        </option>
        <option>
          HARDPROMISED
        </option>
      </select>
    </div>
    <div
      class="form-group col-md-6"
    >
      <label
        data-testid="Quantity"
        for="Quantity"
      >
        Quantity
      </label>
      <input
        class="form-control"
        id="Quantity"
        placeholder="Enter Quantity"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-1u0dfaj-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="submit"
  >
    Upadte Demand
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-7ba66y-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Cancel
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</form>
`;

exports[`renders Demand component correctly 1`] = `
<div
  class="container"
  data-testid="demand"
>
  <div
    class=" mt-3"
  >
    <h3
      class="text-center"
      data-testid="DemandDetails"
      variant="h3"
    >
      Demand Details
    </h3>
  </div>
  <div
    class="row mb-3"
  >
    <div
      class="col-6"
    >
      <div
        class="col-md-6"
      >
        <input
          class="search form-control"
          data-testid="search"
          placeholder="Search Demand By ItemId"
          type="text"
        />
      </div>
    </div>
    <div
      class="col-6 "
    >
      <a
        class="text-decoration-none"
        href="/demand/add?id=-1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium text-black bg-info align-right css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="addDemand"
          tabindex="0"
          type="button"
        >
          Add Demand
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="customized table"
      class="MuiTable-root css-1pclhgl-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            scope="col"
            style="font-size: 17px;"
          >
            Demand Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            scope="col"
            style="font-size: 17px;"
          >
            Item Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            scope="col"
            style="font-size: 17px;"
          >
            Location Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            scope="col"
            style="font-size: 17px;"
          >
            Demand Type
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            scope="col"
            style="font-size: 17px;"
          >
            Quantity
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium css-og52vs-MuiTableCell-root"
            colspan="2"
            scope="col"
            style="font-size: 17px;"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      />
      <tfoot
        class="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeMedium MuiTablePagination-root css-fikjyc-MuiTableCell-root-MuiTablePagination-root"
            colspan="12"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id=":r1:"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
              >
                <select
                  aria-label="rows per page"
                  class="MuiNativeSelect-select MuiTablePagination-select MuiNativeSelect-standard MuiInputBase-input css-m3l1ig-MuiNativeSelect-select-MuiInputBase-input"
                  id=":r0:"
                >
                  <option
                    class="MuiTablePagination-menuItem"
                    value="5"
                  >
                    5
                  </option>
                  <option
                    class="MuiTablePagination-menuItem"
                    value="10"
                  >
                    10
                  </option>
                  <option
                    class="MuiTablePagination-menuItem"
                    value="25"
                  >
                    25
                  </option>
                  <option
                    class="MuiTablePagination-menuItem"
                    value="-1"
                  >
                    All
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiTablePagination-selectIcon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                0â€“0 of 0
              </p>
              <div
                class="MuiBox-root css-1zye22"
              >
                <button
                  aria-label="first page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FirstPageIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="last page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LastPageIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
